src/auth/auth.controller.ts

```diff
@@ -0,0 +1,60 @@
+import { instanceToPlain, plainToInstance } from 'class-transformer';
+import { validate } from 'class-validator';
+import { Request, Response } from 'express';
+
+import { Controller } from '../common';
+
+import { AuthService } from './auth.service';
+import { LoginDto, RegisterDto } from './dto';
+
+export class AuthController extends Controller {
+    constructor() {
+        super('/auth');
+
+        const router = this.getRouter();
+
+        router.post('/login', this.login);
+        router.post('/register', this.register);
+        router.post('/logout', this.logout);
+    }
+
+    async login(req: Request, res: Response) {
+        const body = plainToInstance(LoginDto, req.body as LoginDto);
+        const errors = await validate(body);
+        if (errors.length > 0) {
```

Comment:
just `errors.length`

Reply:
i don't think it is really an improvement. it makes a code harder to read... now it is almost just like english. let me know if you agree

Reply:
It's redundant and its kinda standard to everyone to be aware of how variables and types are working in JS 

Its enough to check if the length property exists

Reply:
ok then